cmd/agent/common/autodiscovery.go

```diff
@@ -265,10 +267,24 @@ func waitForConfigsFromAD(ctx context.Context, wildcard bool, checkNames []strin
 		}
 	}
 
+	stopChan := make(chan struct{})
 	// add the scheduler in a goroutine, since it will schedule any "catch-up" immediately,
 	// placing items in configChan
 	go AC.AddScheduler("check-cmd", schedulerFunc(func(configs []integration.Config) {
 		for _, cfg := range configs {
+			if instanceFilter != "" {
+				instances, err := filterInstances(cfg.Instances, instanceFilter)
+				if err != nil {
+					returnErr = err
```

Comment:
I would prefer to get a list of error, and return an aggregated view of the errors.

we are using in several place https://pkg.go.dev/k8s.io/apimachinery/pkg/util/errors

Reply:
@clamoriniere Thanks for the suggestion. I modified the code to return an aggregated view of the errors here: https://github.com/DataDog/datadog-agent/pull/15034/commits/89534fd924a75b718092fb90df6330b61ed6d786

Let me know if there is anything else to improve.